### PR TITLE
Nodejs week3 meal-sharing

### PR DESCRIPTION
**Pull request by shanhaider23**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -5,11 +5,119 @@ const knex = require("../database");
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    //const titles = await knex("meal_application").select("title");
+    //response.json(titles);
+    await knex("meals").insert({
```
Comment (flom84) [inline src/backend/api/meals.js]: you could generate some meals by using mysql workbench and `database_dump_meals.sql` script

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -5,11 +5,119 @@ const knex = require("../database");
...
+      title: "Egg fried rice",
+      discription: "Egg with rice and some chicken",
+      location: "Glostrup",
+      when_: new Date(),
+      max_reservations: 10,
+      price: 250,
+      created_date: new Date(),
+    });
   } catch (error) {
     throw error;
```
Comment (flom84) [inline src/backend/api/meals.js]: returning an error response is better here than throwing an exception that might crash your server

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -5,11 +5,119 @@ const knex = require("../database");
...
   }
 });
+router.get("/", async (request, response) => {
+  try {
+    if ("maxPrice" in request.query) {
+      const maxPrice = request.query.maxPrice;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
      const maxPrice = Number(request.query.maxPrice);
```
i think you have to be careful when you get a string query parameter and make sure it's a number

File: src/backend/api/meals.js [hunk 4]
```diff
@@ -5,11 +5,119 @@ const knex = require("../database");
...
+      const mealPrice = await knex("meals").where("price", "<=", maxPrice);
+      response.send(mealPrice);
+      return;
+    }
+
+    if ("createdAfter" in request.query) {
+      const createdAfter = request.query.createdAfter;
```
Comment (flom84) [inline src/backend/api/meals.js]: I think you have to do something similar with a date like my comment above with the number. For exemple , you can use `Date.parse(request.query.createdAfter) `

Comment (shanhaider23) [reply src/backend/api/meals.js]: Thanks, I will change it.

**Review by flom84 — APPROVED** [review]

File: src/backend/api/meals.js [hunk 5]
```diff
@@ -4,12 +4,101 @@ const knex = require("../database");
...
+        "title",
+        "like",
+        "%" + title + "%"
+      );
+      response.send(getTitle);
+      return;
+    }
+
+    if ("limit" in request.query) {
+      const limit = request.query.limit;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
      const limit = Number(request.query.limit);
```

File: src/backend/api/meals.js [hunk 6]
```diff
@@ -4,12 +4,101 @@ const knex = require("../database");
...
+    const meals = await knex("meals").insert(request.body);
+    response.json(meals);
+  } catch (error) {
+    throw error;
+  }
+})
+
+router.put("/:id", async (request, response) => {
+  try {
+    const selectedId = request.params.id;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
    const selectedId = Number(request.params.id);
```

File: src/backend/api/meals.js [hunk 7]
```diff
@@ -4,12 +4,101 @@ const knex = require("../database");
...
+    const Selectedmeal = await knex("meals").where("id", selectedId).update(request.body)
+    response.json(Selectedmeal);
+  } catch (error) {
+    throw error;
+  }
+})
+
+router.delete("/:id", async (request, response) => {
+  try {
+    const deletedID = request.params.id;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
    const deletedID = Number(request.params.id);
```

**Review by flom84 — CHANGES_REQUESTED** [review]

**Comment by flom84** [comment]
Make sure that you check all the query parameters and params when you have to retrieve data

**Review by flom84 — APPROVED** [review]

**Pull request merged**